From c0b747666e6c94e0085caf394cef4a13c616d89a Mon Sep 17 00:00:00 2001
From: Ola Jeppsson <ola@adapteva.com>
Date: Mon, 4 Apr 2016 22:08:22 +0200
Subject: [PATCH 30/33] epiphany: Make r32-r43 caller-saved

gcc/
	* config/epiphany/epiphany.h (CALL_USED_REGISTERS): Make
	r32-r43 caller-saved.

Signed-off-by: Ola Jeppsson <ola@adapteva.com>
---
 gcc/ChangeLog.epiphany         | 5 +++++
 gcc/config/epiphany/epiphany.h | 4 ++--
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/gcc/ChangeLog.epiphany b/gcc/ChangeLog.epiphany
index c9ec0920c34..f7c252d7b78 100644
--- a/gcc/ChangeLog.epiphany
+++ b/gcc/ChangeLog.epiphany
@@ -1,3 +1,8 @@
+2016-04-04  Ola Jeppsson  <ola@adapteva.com>
+
+	* config/epiphany/epiphany.h (CALL_USED_REGISTERS): Make
+	r32-r43 caller-saved.
+
 2015-11-02  Ola Jeppsson  <ola@adapteva.com>
 
 	* config/epiphany/epiphany.c (emit_set_fp_mode): If
diff --git a/gcc/config/epiphany/epiphany.h b/gcc/config/epiphany/epiphany.h
index 6a1e1ba6750..86aff0d9f81 100644
--- a/gcc/config/epiphany/epiphany.h
+++ b/gcc/config/epiphany/epiphany.h
@@ -266,8 +266,8 @@ along with GCC; see the file COPYING3.  If not see
 	0, 0, 0, 0, 1, 1, 1, 0,		/* 008-015, gr8  - gr15 */	\
 	1, 1, 1, 1, 1, 1, 1, 1,		/* 016-023, gr16 - gr23 */	\
 	1, 1, 1, 1, 1, 1, 1, 1,		/* 024-031, gr24 - gr31 */	\
-	0, 0, 0, 0, 0, 0, 0, 0,		/* 032-039, gr32 - gr38 */	\
-	0, 0, 0, 0, 1, 1, 1, 1,		/* 040-047, gr40 - gr47 */	\
+	1, 1, 1, 1, 1, 1, 1, 1,		/* 032-039, gr32 - gr38 */	\
+	1, 1, 1, 1, 1, 1, 1, 1,		/* 040-047, gr40 - gr47 */	\
 	1, 1, 1, 1, 1, 1, 1, 1,		/* 048-055, gr48 - gr55 */	\
 	1, 1, 1, 1, 1, 1, 1, 1,		/* 056-063, gr56 - gr63 */	\
 	1,				/* 64 AP   - fake arg ptr */	\
-- 
2.31.1

